qiskit_experiments/framework/base_experiment.py

```diff
@@ -167,29 +144,144 @@ def _initialize_experiment_data(
 
         return experiment_data._copy_metadata()
 
-    def run_analysis(self, experiment_data, **options) -> ExperimentData:
+    def pre_transpile_action(self, backend: Backend):
+        """An extra subroutine executed before transpilation.
+
+        Note:
```

Comment:
This doc string will print for every experiment subclass that doesn't override this method which will look pretty odd. The doc string seems to make more sense as code comment or base class docs.

Reply:
That was from this comment https://github.com/Qiskit/qiskit-experiments/pull/380#discussion_r706863232. I'm fine with either way, but considering documentation protected method seems to be better approach.

Reply:
Done in https://github.com/Qiskit/qiskit-experiments/pull/380/commits/e234afa16964d13faa9ad951af1cd9643c7a55d0